src/backend/cuda/convolve.cpp

```diff
@@ -39,14 +39,20 @@ template<>
 cudnnDataType_t getCudnnDataType<double>() {
     return CUDNN_DATA_DOUBLE;
 }
+
+#if CUDNN_MAJOR >= 6
 template<>
 cudnnDataType_t getCudnnDataType<int>() {
     return CUDNN_DATA_INT32;
 }
+
+#elif CUDNN_MAJOR > 7 || (CUDNN_MAJOR >= 7 && CUDNN_MINOR >= 1)
```

Comment:
We will never reach the code-path under this condition. I think you were aiming for a separate `#if` here.